### DISCUSSION TITLE
Feature request to mitigate race conditions risks

### DISCUSSION THREAD

**Discussion by Voldemorten** [directus/directus#11667] [Feature Requests]
Hi Directus community!

I don't see anywhere, that there's any handling of potential race conditions. My suggestion (and @MiniDigger's ) for a mitigation is two fold:
1. Tell you if somebody else is viewing the same item
2. Prevent saving if somebody else saved the item after you read

What do you think?

**Comment by benhaynes** [comment]
I think we have this in the works, right @rijkvanzanten ?

**Comment by rijkvanzanten** [comment]
I'm in favor! And yes, that's something we have in the works. We're currently looking into WebSocket support in the API, which in turn would allow the app to easily receive these messages when they happen. This in turn would allow us to render warnings, disable save, make fields readonly, and even potentially live update the item whenever somebody else saves it 🙂

**Reply by Voldemorten** [reply]
Sounds awesome!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
